google-cloud-storage/src/main/java/com/google/cloud/storage/OtelStorageDecorator.java

```diff
@@ -1650,21 +1655,38 @@ public boolean isOpen() {
 
     @Override
     public void close() {
+      setScope();
       try {
         reader.close();
       } finally {
         span.end();
+        clearScope();
+      }
+    }
+
+    private void clearScope() {
+      try (Scope ignore = scope) {
+        scope = null;
+      }
+    }
+
+    public void setScope() {
+      if (scope != null) {
+        clearScope();
       }
+      scope = span.makeCurrent();
     }
```

Comment:
This looks great! TIL to have explicit scope handling added to properly activate spans for lazy RPCs